dpctl/_sycl_device.pyx

```diff
@@ -734,10 +740,17 @@ cdef class SyclDevice(_SyclDevice):
             )
         for i in range(ncounts):
             counts_buff[i] = counts[i]
-        DVRef = DPCTLDevice_CreateSubDevicesByCounts(
-            self._device_ref, counts_buff, ncounts
-        )
+            if counts_buff[i] == 0:
```

Comment:
Why not do this before you `malloc` counts_buff to not even allocate the array if input is not valid.